src/vs/base/test/browser/comparers.test.ts

```diff
@@ -3,48 +3,556 @@
+
+		// mixed-case dotfile comparisons
+		assert(compareFileExtensionsMixed('A.MD', '.md') > 0, 'dotfiles sort before uppercase files');
+		assert(compareFileExtensionsMixed('a.md', '.MD') > 0, 'dotfiles sort before lowercase files');
+		assert(compareFileExtensionsMixed('.md', '.Gitattributes') > 0, 'dotfiles sort alphabetically regardless of case');
+
+		// same-case numeric comparisons
+		assert(compareFileExtensionsMixed('abc2.txt', 'abc10.txt') > 0, 'filenames with numbers should be in alphabetical order');
+		assert(compareFileExtensionsMixed('abc02.txt', 'abc002.txt') > 0, 'filenames with equivalent numbers and leading zeros sort alphabetically');
+		assert(compareFileExtensionsMixed('abc02.txt', 'abc010.txt') > 0, 'filenames with numbers that have leading zeros sort alphabetically');
+		assert(compareFileExtensionsMixed('abc02.txt', 'abc010.txt') > 0, 'filenames with numbers that have leading zeros sort alphabetically');
+		assert(compareFileExtensionsMixed('abc1.10.txt', 'abc1.2.txt') < 0, 'numbers with dots between sort alphabetically');
+
+		// mixed-case numeric comparisons
+		assert(compareFileExtensionsMixed('art01', 'Art01') < 0, 'a numerically equivalent word of a different case sorts lowercase first');
+
+		// comparisons that depend on comparing extensions then names
+		assert(compareFileExtensionsMixed('bbb.aaa', 'aaa.bbb') < 0, 'files should be compared by extension first');
+		assert(compareFileExtensionsMixed('agg.go', 'aggrepo.go') < 0, 'shorter names sort before longer names');
+		assert(compareFileExtensionsMixed('agg.go', 'agg_repo.go') < 0, 'shorter names short before longer names even when the longer name contains an underscore');
+		assert(compareFileExtensionsMixed('abc2.04txt', 'abc2.2txt') < 0, 'numbers around a the extension dot separator are not compared alphabetically');
+		assert(compareFileExtensionsMixed('abc.02txt', 'abc.2txt') < 0, 'equivalent numbers in extensions sort alphabetically');
+
+		// comparisons that depend on how extensions plus case are handled
+		assert(compareFileExtensionsMixed('a.MD', 'a.md') === 'MD'.localeCompare('md'), 'case differences in extensions sort by locale');
+		assert(compareFileExtensionsMixed('a.MD', 'b.md') < 0, 'when extensions are the same except in case, the files sort by name');
+		assert(compareFileExtensionsMixed('art.MD', 'Art.MD') === 'art'.localeCompare('Art'), 'when extensions are the same, files sort by name and locale');
+		assert(compareFileExtensionsMixed('art.MD', 'Art.md') === 'art'.localeCompare('Art'), 'when extensions are the same except in case, files sort by name and locale');
+
+		// comparisons that depend on how case and numbers are handled
+		assert(compareFileExtensionsMixed('a10.txt', 'A2.txt') === 'a10'.localeCompare('A2'), 'filenames with numbers and case differences sort by locale');
+
 	});
+
+	test('compareCaseLowerFirst', () => {
```

Comment:
I can remove these tests and not export these functions if that's preferred. The compareCase* functions are helper functions. It was good to be able to test them directly during development, but the other tests cover their behavior indirectly.